Library/ViewModels/PaymentMethodsViewModelTests.swift

```diff
@@ -197,6 +197,70 @@ internal final class PaymentMethodsViewModelTests: TestCase {
     }
   }
 
+  func testDeletePaymentMethod_OptimisticDelete() {
+    let apiService = MockService(deletePaymentMethodResult: .success(.init(totalCount: 2)))
+    withEnvironment(apiService: apiService) {
+
+      self.vm.inputs.viewWillAppear()
+      self.scheduler.advance()
+
+      self.tableViewIsEditing.assertValues([])
+      self.showAlert.assertValues([])
+      self.paymentMethods.assertValues([GraphUserCreditCard.template.storedCards.nodes])
+
+      self.vm.inputs.editButtonTapped()
+
+      self.tableViewIsEditing.assertValues([true], "Editing mode enabled")
+      self.showAlert.assertValues([])
+      self.paymentMethods.assertValues([GraphUserCreditCard.template.storedCards.nodes])
+
+      GraphUserCreditCard.template.storedCards.nodes.forEach { [weak self] card in
+        self?.vm.didDelete(card)
+      }
+
+      self.editButtonIsEnabled.assertValues([true, false], "Editing mode disabled, all cards deleted")
+
+      // advance scheduler to perform delete call
+      self.scheduler.advance()
+
+      self.editButtonIsEnabled.assertValues(
+        [true, false, true], "Editing mode enabled, delete response has count of 2"
```

Comment:
So in this scenario although all cards were initially deleted, a few calls "failed" and thats why the edit button is re-enabled, correct? How is that different from the test below?

Reply:
Can we add a test where the button stays disabled when all cards are successfully deleted?